index.bs

```diff
@@ -5342,6 +5362,93 @@ over just using {{SyntaxError!!exception}} to refer to the {{DOMException}}. [[D
+
+*   The [=identifier=] of the [=interface=] must end with <code>Error</code>, and must not be any
+    of the names in the <a><code>DOMException</code> names table</a>.
+*   The [=interface=] must have a [=constructor operation=] which sets the instance's
+    [=DOMException/name=] to the interface's [=identifier=].
+*   Their [=constructor operation=] must take as its first parameter an [=optional=] {{DOMString}}
+    named |message| defaulting to the empty string, and must set the instance's
+    [=DOMException/message=] to |message|.
+*   Their [=constructor operation=] should take as its second parameter a [=dictionary=] containing
+    the additional information that needs to be exposed.
+*   They should have [=read only=] [=attributes=], whose names are the same as the members of the
+    constructor dictionary, which return the values accepted by the constructor operation.
+*   They should be [=serializable objects=], whose [=serialization steps=] and
+    [=deserialization steps=] preserve the additional information.
+
+<p class=note>These requirements mean that the inherited {{DOMException/code}} property of these
+interfaces will always return 0.
+
+<div class=example id=example-domexception-derived-interface>
+    The definition for a {{DOMException}} derived interface which carries along an additional
+    "protocol error code", which is derived from what the server sent over some some hypothetical
+    network protocol "protocol X", could look something like this:
+
+    <pre highlight=webidl>
+        [Exposed=Window, Serializable]
+        interface ProtocolXError : DOMException {
+          constructor(optional DOMString message = "", ProtocolXErrorOptions options);
+
+          readonly attribute unsigned long long errorCode;
+        };
+
+        dictionary ProtocolXErrorOptions {
+            required [EnforceRange] unsigned long long errorCode;
+        };
+    </pre>
+
+    Every <code>ProtocolXError</code> instance has an <dfn for="ProtocolXError">error code</dfn>,
+    a number.
+
+    <div algorithm="ProtocolXError constructor">
+        The <b><code>new ProtocolXError(|message|, |options|)</code></b> constructor steps are:
+
+        1.  Set [=this=]'s [=DOMException/name=] to "<code>ProtocolXError</code>".
+        1.  Set [=this=]'s [=DOMException/message=] to |message|.
+        1.  Set [=this=]'s [=ProtocolXError/error code=] to |options|["<code>errorCode</code>"].
+    </div>
+
+    <div algorithm="ProtocolXError errorCode">
+        The <b><code>errorCode</code></b> getter steps are to return [=this=]'s
+        [=ProtocolXError/error code=].
+    </div>
+
+    <code>ProtocolXError</code> objects are [=serializable objects=].
+
+    <div algorithm="ProtocolXError serialization steps">
+        Their [=serialization steps=], given |value| and |serialized|, are:
+
+        1. Run the {{DOMException}} [=serialization steps=] given |value| and |serialized|.
```

Comment:
Nit: not sure what best practice is, but it seems odd to me to use "given" both on the calling and receiving end of an algorithm. I'd expect "with" on the calling end, like [two lines down from here](https://dom.spec.whatwg.org/#set-the-canceled-flag).

```suggestion
        Their [=serialization steps=], given |value| and |serialized|, are:

        1. Run the {{DOMException}} [=serialization steps=] with |value| and |serialized|.
```

Reply:
"given" is fine here.